build_resolvers/lib/src/resolver.dart

```diff
@@ -402,30 +403,65 @@ class AnalyzerResolvers implements Resolvers {
   /// **NOTE**: The [_packageConfig] is not used for path resolution, it is
   /// primarily used to get the language versions. Any other data (including
   /// extra data), may be passed to the analyzer on an as needed basis.
-  AnalyzerResolvers(
-      [AnalysisOptions? analysisOptions,
-      Future<String> Function()? sdkSummaryGenerator,
-      this._packageConfig])
-      : _analysisOptions = analysisOptions ??
+  factory AnalyzerResolvers.custom({
+    AnalysisOptions? analysisOptions,
+    Future<String> Function()? sdkSummaryGenerator,
+    PackageConfig? packageConfig,
+  }) =>
+      AnalyzerResolvers._(
+          analysisOptions: analysisOptions,
+          sdkSummaryGenerator: sdkSummaryGenerator,
+          packageConfig: packageConfig,
+          // Custom resolvers get their own asset uri resolver, as there should
+          // always be a 1:1 relationship between them.
+          uriResolver: BuildAssetUriResolver());
+
+  /// See [AnalyzerResolvers.custom] for docs.
```

Comment:
[optional] Could use a `{@template analyzer_resolvers_constructor}` and `{@macro analyzer_resolvers_constructor}` to copy the doc to the other constructors.

Reply:
Looks like you lose navigation from within the IDE though so I think that is ultimately worse?